mathics/core/definitions.py

```diff
@@ -725,25 +725,122 @@ def get_history_length(self):
 
 
 def get_tag_position(pattern, name) -> Optional[str]:
+    """
+    Determine the position of a pattern in
+    the definition of the symbol ``name``
+    """
+    blanks = (
+        "System`Blank",
+        "System`BlankSequence",
+        "System`BlankNullSequence",
+    )
+
+    def strip_pattern_name_and_condition(pat):
+        """
+        In ``Pattern[name_, pattern_]`` and
+        ``Condition[pattern_, cond_]``
+        the tag is determined by pat.
+        This function strips it to ensure that
+        ``pat`` does not have that form.
+        """
+        if pat.get_head_name() == "System`Condition":
+            if len(pat.elements) > 1:
+                return strip_pattern_name_and_condition(pat.elements[0])
+        if pat.get_head_name() == "System`Pattern":
+            if len(pat.elements) == 2:
+                return strip_pattern_name_and_condition(pat.elements[1])
+        return pat
+
+    def check_is_subvalue(pattern_sv, name_sv):
```

Comment:
Add annotation for function signature: 

```
def check_is_subvalue(pattern_sv: ExpressionPattern, name_sv: str) -> bool:
```

the function name `check_is_subvalue()` is a poor name. First, for functions returning boolean values we use `is_xxx` not `check_is_xxx`. Second what is a subvalue? 

In the PR that will appear I have changed the docstring to: 

```
"""Determines if ``pattern_sv`` or any of its alternates is a pattern with name `name_sv`"""
```

Here

Reply:
In the PR I have changed the function name to "is_pattern_a_kind_of".